malcolm/core/numbermeta.py

```diff
@@ -0,0 +1,40 @@
+from malcolm.core.attributemeta import AttributeMeta
+
+from pkg_resources import require
+require("numpy")
+import numpy as np
+
+
+_metas = {
+    np.int32:"malcolm:core/Int:1.0",
+    np.float64:"malcolm:core/Float:1.0"
+    }
+
+@AttributeMeta.register_subclass(_metas[np.int32], np.int32)
+@AttributeMeta.register_subclass(_metas[np.float64], np.float64)
+class NumberMeta(AttributeMeta):
+    """Meta object containing information for a numerical value"""
+
+    def __init__(self, name, description, dtype):
+        super(NumberMeta, self).__init__(name=name, description=description)
+        self.dtype = dtype
+        # TODO where and when should this be set?
```

Comment:
I think the base class should do this. If we maintain the inverse mapping from (subcls, args) -> metaOf is AttributeMeta.register_subclass, then AttributeMeta.__init__(name, description, *args) can set self.metaOf based on this information